pinot-core/src/main/java/org/apache/pinot/core/operator/combine/CombineOperatorUtils.java

```diff
@@ -65,13 +68,24 @@ public static void setExecutionStatistics(IntermediateResultsBlock resultsBlock,
       if (executionStatistics.getNumDocsScanned() > 0) {
         numSegmentsMatched++;
       }
+      try {
+        if (operator.getIndexSegment() instanceof MutableSegment) {
+          numConsumingSegmentsProcessed += 1;
+          if (executionStatistics.getNumDocsScanned() > 0) {
+            numConsumingSegmentsMatched++;
+          }
+        }
+      } catch (UnsupportedOperationException ignored) { }
```

Comment:
This try catch is redundant

Reply:
`SelectionCombineOperatorTest.testSelectionLimit0` Should I modify this test to expect an exception as well ? Also while I was trying the query I was getting UnsupportedException in the response but the exception has happened while fetching the metadata. So it gives an impression that the query is not supported which it is.

Let me know if you still feel we want to change this, Thanks

Reply:
I see. All the segment top level operator should have this implemented, or this won't be accurate. Seems it is missing from several operators. We may keep it like this for now, and let's add a TODO to check all operators and make sure they have that method properly implemented in a separate PR.

(nit) We usually use 2 lines even when the block is empty
```suggestion
      } catch (UnsupportedOperationException ignored) {
      }
```